gn2/wqflask/views.py

```diff
@@ -268,11 +268,47 @@ def gsearchtable():
 
     return flask.jsonify(current_page)
 
+
 def clean_xapian_query(query: str) -> str:
-    """ Remove filler words in xapian query
-    TODO: FIXME
     """
-    return query
+    Clean and optimize a Xapian query string by removing filler words,
+    and ensuring the query is tailored for optimal results from Fahamu.
+
+    Args:
+        query (str): The original Xapian query string.
+
+    Returns:
+        str: The cleaned and optimized query string.
+    """
+    xapian_prefixes = {
+        "author",
+        "species",
+        "group",
+        "tissue",
+        "dataset",
+        "symbol",
+        "description",
+        "rif",
+        "wiki",
+    }
+    xapian_operators = {"AND", "NOT", "OR", "XOR", "NEAR", "ADJ"}
+    range_prefixes = {"mean", "peak", "position", "peakmb", "additive", "year"}
+    query_context = ["genes"]
+    cleaned_query_parts = []
+    for token in query.split():
+        if token in xapian_operators or any(
+            prefix in token for prefix in range_prefixes if ".." in token
+        ):
+            continue
+        prefix, _, suffix = token.partition(":")
+        if prefix in xapian_prefixes:
+            query_context.insert(0, prefix)
+            cleaned_query_parts.append(f"{prefix} {suffix}")
+        else:
+            cleaned_query_parts.append(prefix)
+    cleaned_query = " ".join(cleaned_query_parts)
+    context = ",".join(query_context)
+    return f"Provide answer on {cleaned_query} context {context}"
```

Comment:
Question: 
1. Does this results in better results and should we somehow integrate this in the normal searches?
2. Should we add some type of formatting to the text i.e.
```bash
# if I search for:
rif:glioma species:human group:gtex_v8
# we pass in the query:
Provide answer on glioma human gtex_v8 context rif,species,group
# I thought we'd add some identifiers like
Provide answer on `glioma human gtex_v8` with context `[rif, species,group]`
```

Reply:
1) Yeah, I conducted some experiments on this and others. It reduces the amount of 
"No feedback response/text does not contain".... responses we get
 also provides  more information on the subject with context provided
 Ideally Fahamu should be able to pick few keywords and get a response
 but currently it doesn't If this improves the normal search also, that would be okay
2) I would need to test this.